src/backends/policy_engine/souffle/check_policy_compliance.cc

```diff
@@ -27,11 +27,14 @@
 #include "src/backends/policy_engine/policy.h"
 #include "src/backends/policy_engine/souffle/souffle_policy_checker.h"
 #include "src/common/logging/logging.h"
+#include "src/ir/proto_to_ir.h"
 #include "src/parser/ir/ir_parser.h"
 
-ABSL_FLAG(std::string, ir, "", "the IR file");
+// TODO(cromwellian): why doesn't std::optional<std::string> (or absl::optional) work?
```

Comment:
What error do you have when using `std::optional<std::string>`?

Reply:
I fixed it, see the discussion in the Raksha chat. This was caused by an old version ABSL, I upgraded our workspace to pickup the newest version that supports optional flags. 

(The error you'll get is the same you'll get for any custom type: Can't find AbslUnparseFlags(TYPE). This is because marshalling.h didn't provide an unparser for std::optional out of the box in older versions)

Reply:
SG, please remove the TODOs.